components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java

```diff
@@ -351,6 +354,13 @@ protected boolean handlePostAuthentication(Event event, String userName, UserSto
                     throw new AccountLockException(USER_IS_LOCKED + ":" + AccountConstants.ADMIN_INITIATED, message);
                 }
 
+                if (MAX_ATTEMPTS_EXCEEDED.toString().equals(accountLockedReason)) {
```

Comment:
why we nee toString() here and other places, doesn't them are string constants

Reply:
These are enums. When storing as the value for locked reason claim, the enum is converted to its string representation and stored in the DB. Therefore, here also we have to check the equality with the string representation of the enum

Reply:
I was refering to these constants. https://github.com/wso2-extensions/identity-event-handler-account-lock/pull/123/files#diff-0322b1e25beb96d910d694060cac8007e2173ff2dd95664ae0a050c65bf06a08R73 
Can't we use them.

Reply:
These are different from the values we store in DB for locked account claim. Values stored in DB for locked account claim are in the form MAX_ATTEMPTS_EXCEEDED where as values we show to the user are in PascalCase (i.e. MaxAttemptsExceeded). Here the check is done for the value stored in DB.

The confusion seems to be the identical name used in defining the enum and the string constant. Can change the string constant name for the MAX_ATTEMPTS_EXCEEDED case (which I have introduced newly). However, for ADMIN_INITIATED case, this cannot be done as it is an already existing public constant, which might have been used elsewhere.